linetools/isgm/dla.py

```diff
@@ -0,0 +1,247 @@
+""" Subclasses for DLA AbsSystem and AbsSurvey
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import os
+import pdb
+
+from astropy import units as u
+
+from linetools.isgm.abssystem import AbsSystem
+from linetools.isgm.abssurvey import AbslineSurvey
+from linetools.isgm import utils as ltiu
+
+
+class DLASystem(AbsSystem):
+    """
+    Class for a DLA absorption system
+    """
+    @classmethod
+    def from_datfile(cls, dat_file, tree=None, **kwargs):
+        """ Read from dat_file (historical JXP format)
+
+        Parameters
+        ----------
+        dat_file : str
+          dat file
+        tree : str, optional
+          Path to data files
+        kwargs :
+          Passed to __init__
+
+        Returns
+        -------
+        _datdict : dict
+          Fills this attribute
+        """
+        if tree is None:
+            tree = ''
+        # Read datfile
+        datdict = ltiu.read_dat_file(tree+dat_file)
+        # Parse
+        coord, zabs, name, NHI, sigNHI, clm_fil = ltiu.parse_datdict(datdict)
+        kwargs['NHI'] = NHI
+        kwargs['sig_NHI'] = sigNHI
+
+        # Generate with type
+        vlim = None
+        slf = cls(coord, zabs, vlim, **kwargs)
+
+        # Fill files
+        slf.tree = tree
+        slf.dat_file = slf.tree+dat_file
+
+        # Parse datdict
+        slf._datdict = datdict
+
+        # QSO keys
+        slf.qso = slf._datdict['QSO name']
+        slf.zqso = float(slf._datdict['QSO zem'])
+        # Name
+        slf.name = '{:s}_z{:0.3f}'.format(slf.qso,zabs)
+
+        # Abund
+        slf.flg_ZH = float(slf._datdict['flg_mtl'])
+        slf.ZH = float(slf._datdict['[M/H]'])
+        slf.sig_ZH = float(slf._datdict['sig([M/H])'])
+
+        return slf
+
+    def __init__(self, radec, zabs, vlim, **kwargs):
+        """Standard init
+
+        NHI keyword is required
```

Comment:
Why isn't it a positional parameter like to others, then? You can still pass positional parameters as keywords if you want to do this just to be explicit. e.g.

``` python
def f(a,b): print(a, b)
f(1, 2)
f(b=2, a=1)
```
do the same thing